File: app/src/main/java/com/ruuvi/station/adapters/AddTagAdapter.java
package com.ruuvi.station.adapters;

import android.content.Context;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.widget.AppCompatImageView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import com.ruuvi.station.R;
import com.ruuvi.station.database.tables.RuuviTagEntity;

import java.util.List;

/**
 * Created by berg on 10/10/17.
 */

public class AddTagAdapter extends ArrayAdapter<RuuviTagEntity> {

    private List<RuuviTagEntity> tags;

    public AddTagAdapter(@NonNull Context context, List<RuuviTagEntity> tags) {
        super(context, 0, tags);
        this.tags = tags;
    }

    @NonNull
    @Override
    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
        final RuuviTagEntity tag = getItem(position);

        if (convertView == null) {
            convertView = LayoutInflater.from(getContext()).inflate(R.layout.row_item_add, parent, false);
        }

        ((TextView)convertView.findViewById(R.id.address)).setText(tag.getId());
        ((TextView)convertView.findViewById(R.id.rssi)).setText(String.format(getContext().getResources().getString(R.string.signal_reading), tag.getRssi()));

        AppCompatImageView signalIcon = convertView.findViewById(R.id.signalIcon);
        if (tag.getRssi() < -80) signalIcon.setImageResource(R.drawable.icon_connection_1);
        else if (tag.getRssi() < -50) signalIcon.setImageResource(R.drawable.icon_connection_2);
        else signalIcon.setImageResource(R.drawable.icon_connection_3);

        return convertView;
    }
}

File: app/src/main/java/com/ruuvi/station/addtag/ui/AddTagAdapter.kt
package com.ruuvi.station.addtag.ui

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import com.ruuvi.station.R
import com.ruuvi.station.database.tables.RuuviTagEntity
import com.ruuvi.station.databinding.RowItemAddBinding
import com.ruuvi.station.util.extensions.diffGreaterThan

class AddTagAdapter(
    context: Context,
    items: List<RuuviTagEntity>
) : ArrayAdapter<RuuviTagEntity>(context, 0, items) {
    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val item = getItem(position)

        val binding = if (convertView != null) {
            RowItemAddBinding.bind(convertView)
        } else {
            RowItemAddBinding.inflate(LayoutInflater.from(context), parent, false)
        }

        with(binding) {
            address.text = item?.displayName()
            rssi.text = context.getString(
                R.string.signal_reading,
                item?.rssi,
                context.getString(R.string.signal_unit)
            )

            when {
                item?.rssi?.compareTo(LOW_SIGNAL) == -1 -> signalIcon.setImageResource(R.drawable.icon_connection_1)
                item?.rssi?.compareTo(MEDIUM_SIGNAL) == -1 -> signalIcon.setImageResource(R.drawable.icon_connection_2)
                else -> signalIcon.setImageResource(R.drawable.icon_connection_3)
            }

            address.isEnabled = item?.updateAt?.diffGreaterThan(10000) != true

            return root
        }
    }

    companion object {
        private const val LOW_SIGNAL = -80
        private const val MEDIUM_SIGNAL = -50
    }
}